ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -932,40 +932,40 @@ CURLcode HttpOperation::Setup()
       return rc;
     }
 
-    rc = SetCurlPtrOption(CURLOPT_WRITEFUNCTION, (void *)&HttpOperation::WriteMemoryCallback);
+    rc = SetCurlPtrOption(CURLOPT_WRITEFUNCTION, reinterpret_cast<void *>(&HttpOperation::WriteMemoryCallback));
```

Comment:
For all the calls to `SetCurlPtrOption`, the code was using a cast because a macro was used in the past.

Now that `SetCurlPtrOption` is a real function that takes a typed `void *` parameter, I think all the casts can be removed instead.

Reply:
At least on this specific line that results in:

```
ext/src/http/client/curl/http_operation_curl.cc:935:50: error: cannot initialize a parameter of type 'void *' with an rvalue of type 'size_t (*)(void *, size_t, size_t, void *)' (aka 'unsigned long (*)(void *, unsigned long, unsigned long, void *)')
    rc = SetCurlPtrOption(CURLOPT_WRITEFUNCTION, &HttpOperation::WriteMemoryCallback);
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ext/src/http/client/curl/http_operation_curl.cc:509:67: note: passing argument to parameter 'value' here
CURLcode HttpOperation::SetCurlPtrOption(CURLoption option, void *value)
                                                                  ^
```

Reply:
This specific one may need a cast, but others can be simplified I hope.

Reply:
this is true for the `static_cast`s in this file though, so I removed those